algorithms/src/main/kotlin/io/kinference/algorithms/completion/generation/search/BeamSearch.kt

```diff
@@ -76,38 +70,30 @@ internal class BeamSearch(
         }
     }
 
-    override fun currentHypotheses(): List<List<GenerationInfo>> {
-        getNormalizedScores().apply { for (i in this.indices) this[i] = exp(this[i]) }
-        val ans = List(hypotheses.size) { GenerationInfo(eachStepProbs[it], hypotheses[it]) }
-
-        return listOf(ans)
+    override fun hypotheses(): List<MutableList<Int>> {
+        return hypotheses
     }
 
     override fun lastPredictions(): IntArray {
         assert(hypotheses.isNotEmpty() && hypotheses[0].size > 0) { "Can't get last predictions if no steps have been performed" }
         return IntArray(hypotheses.size) { hypotheses[it].last() }
     }
 
+    override fun scores(): MutableList<Double> {
+        return scores
```

Comment:
Same :) If you are just returning variable -- make it one line of code. Also, are you sure that you don't want to make getter of `scores` public and setter private?